Khatib amjad/issue56

khatibAmjad opened the pull request:
issue #56 
1. Moved errorHandler function to utils file
2. Moved the authentication middleware to the guard file inside the middleware folder
3. Fixed the public paths to work in the authentication middleware
4. Deleted the database url from the console.log in the connection file
5. Moved redirect uris from passport.js to .env file
6. Moved swagger definition and options from app.js to the constants file

khatibAmjad committed:
test

khatibAmjad committed:
edit package.json

khatibAmjad committed:
edit yarn.lock

khatibAmjad committed:
edit package.json

khatibAmjad committed:
edit port

khatibAmjad committed:
Merge branch 'dev' of github.com:RCDD-202203-TUR-BEW/backend-capstone-turkey-lets-share into dev

khatibAmjad committed:
add new dev to heroku

khatibAmjad committed:
change swagger server

khatibAmjad committed:
change callbackURL

khatibAmjad committed:
1. move errorHandler to utils file
2. move the authentication middleware to the guard file inside the middleware folder
3. fix the public paths to work in the authentication middleware
4. delete the database url from the console.log in the connection file
5. move redirect uris from passport.js to .env file
6. move swagger definition and options from app.js to the constants file

awiednoor reviewed (commented):
Hey Amjad , can we also edit the swagger paths tags so similar routes are unified under one tag, smth similar to this 
<img width="1508" alt="Screen Shot 2022-08-17 at 15 31 21" src="https://user-images.githubusercontent.com/40646093/185123619-40694f4c-c298-45a6-899e-4648461603bd.png">
e

khatibAmjad committed:
fix typo in script

khatibAmjad committed:
comment out middleware

khatibAmjad committed:
fix middleware path

khatibAmjad committed:
merge updated dev branch

khatibAmjad committed:
1. Add test for authentication middleware
2. delete extra eslint comments
3. fix catch block format in product.js
4. remove /api/user/:userId/products route from public routes in the constants file
5. move the regestering validation middleware to the validation file
6. delete auth.js file from the middleware folder
7. rearrange routes in the auth.js file

khatibAmjad committed:
fix typo

khatibAmjad committed:
delete old auth middleware

awiednoor commented on src/services/utils.js:
```diff
@@ -3,4 +3,19 @@ const generateUniqeUsername = (email) => {
+  }
+}
+
+module.exports = { generateUniqeUsername, errorHandler };
```
Can we add return statement to avoid any errors

khatibAmjad replied:
It's unnecessary since we have if and else written explicitly.

khatibAmjad committed:
add cors

khatibAmjad committed:
add cors package

khatibAmjad committed:
change cors config

khatibAmjad committed:
edit cors options

khatibAmjad committed:
delete cors options

khatibAmjad committed:
merge updated dev branch
